Multisig/UI/App/UIKit/TransactionlListViewController/TransactionListViewController.swift

```diff
@@ -144,88 +165,176 @@ class TransactionListViewController: LoadableViewController, UITableViewDelegate
-            return .awaitingExecution
-        case .cancelled:
-            return .cancelled
-        case .failed:
-            return .failed
-        case .success:
-            return .success
-        case .pending:
-            return .pending
+            return cell
+        case .label(let label):
+            let cell = tableView.dequeueCell(TransactionListHeaderTableViewCell.self, for: indexPath)
+            cell.titleLabel.text = label.label
+
+            return cell
+        case .transaction(let transaction):
+            let cell = tableView.dequeueCell(TransactionListTableViewCell.self, for: indexPath)
+            configure(cell: cell, transaction: transaction)
+
+            return cell
+        case .unknown:
+            return UITableViewCell()
+        }
+    }
+
+    func configure(cell: TransactionListTableViewCell, transaction: SCG.TransactionSummaryItemTransaction) {
+        let tx = transaction.transaction
+        var title = ""
+        var image = #imageLiteral(resourceName: "ico-settings-tx")
+
+        let nonce = tx.executionInfo?.nonce.description ?? ""
+        let confirmationsSubmitted = tx.executionInfo?.confirmationsSubmitted ?? 0
+        let confirmationsRequired = tx.executionInfo?.confirmationsRequired ?? 0
+        let date = formatted(date: tx.timestamp)
+        var info = ""
+
+        var status: SCG.TxStatus = tx.txStatus
+        let missingSigners = tx.executionInfo?.missingSigners?.map { $0.address.checksummed } ?? []
+        if let signingKeyAddress = App.shared.settings.signingKeyAddress,status == .awaitingConfirmations {
+            if missingSigners.contains(signingKeyAddress) {
+                status = .awaitingYourConfirmation
+            }
+        }
+
+        switch transaction.transaction.txInfo {
+        case .transfer(let transferInfo):
+            let isOutgoing = transferInfo.direction == .outgoing
+            image = isOutgoing ? #imageLiteral(resourceName: "ico-outgoing-tx") : #imageLiteral(resourceName: "ico-incoming-tx")
+            title = isOutgoing ? "Send" : "Receive"
+            info = formattedAmount(transferInfo: transferInfo)
+        case .settingsChange(let settingsChangeInfo):
+            title = settingsChangeInfo.dataDecoded.method
+            image = #imageLiteral(resourceName: "ico-settings-tx")
+        case .custom(let customInfo):
+            title = "Contract interaction"
+            info = customInfo.methodName ?? ""
+            image = #imageLiteral(resourceName: "ico-custom-tx")
+        case .creation(_):
+            image = #imageLiteral(resourceName: "ico-settings-tx")
+            title = "Safe created"
+        case .unknown:
+            image = #imageLiteral(resourceName: "ico-custom-tx")
+            title = "Unknown operation"
         }
+
+        cell.set(title, image: image, status: status, nonce: nonce, date: date, info: info, confirmationsSubmitted: confirmationsSubmitted, confirmationsRequired: confirmationsRequired)
```

Comment:
I think it would read better if to have smaller methods like 
cell.set(title: "")
cell.set(image: #imageLiteral(resourceName: "ico-settings-tx"))
...

Reply:
I did it like this because all those parameters are set together at once so there might be no added value of having separated methods in this case

Reply:
I see your point. I just remember that it was explicit Dima's preference in having many small methods per cell property while reviewing my PRs.